src/io/image_aug_default.cc

```diff
@@ -218,10 +265,93 @@ class DefaultImageAugmenter : public ImageAugmenter {
+              }
+              if (y_area <= res.rows && x_area <= res.cols) {
+                // random crop
+                index_t rand_y_area =
+                    std::uniform_int_distribution<index_t>(0, res.rows - y_area)(*prnd);
+                index_t rand_x_area =
+                    std::uniform_int_distribution<index_t>(0, res.cols - x_area)(*prnd);
+                cv::Rect roi(rand_x_area, rand_y_area, x_area, y_area);
+                int interpolation_method = GetInterMethod(param_.inter_method, x_area, y_area,
+                                                          param_.data_shape[2],
+                                                          param_.data_shape[1], prnd);
+                cv::resize(res(roi), res, cv::Size(param_.data_shape[2], param_.data_shape[1]),
+                           0, 0, interpolation_method);
+                attemp = true;
+                break;
+              }
+            }
+            if (!attemp) {
+              // center crop
+              int interpolation_method = GetInterMethod(param_.inter_method, res.cols, res.rows,
+                                                        param_.data_shape[2],
+                                                        param_.data_shape[1], prnd);
+              if (res.rows < param_.data_shape[1]) {
+                index_t new_cols = static_cast<index_t>(static_cast<float>(param_.data_shape[1]) /
+                                                        static_cast<float>(res.rows) *
+                                                        static_cast<float>(res.cols));
+                cv::resize(res, res, cv::Size(new_cols, param_.data_shape[1]),
+                           0, 0, interpolation_method);
+              }
+              if (res.cols < param_.data_shape[2]) {
+                index_t new_rows = static_cast<index_t>(static_cast<float>(param_.data_shape[2]) /
+                                                        static_cast<float>(res.cols) *
+                                                        static_cast<float>(res.rows));
+                cv::resize(res, res, cv::Size(param_.data_shape[2], new_rows),
+                           0, 0, interpolation_method);
+              }
+              CHECK(static_cast<index_t>(res.rows) >= param_.data_shape[1]
+                    && static_cast<index_t>(res.cols) >= param_.data_shape[2])
+                    << "input image size smaller than input shape";
+              index_t center_y = res.rows - param_.data_shape[1];
+              index_t center_x = res.cols - param_.data_shape[2];
+              center_y /= 2;
+              center_x /= 2;
+              cv::Rect roi(center_x, center_y, param_.data_shape[2], param_.data_shape[1]);
+              res = res(roi);
+            }
+            random_resized_crop_exec = true;
+      }
+    }
     // normal augmentation by affine transformation.
     if (param_.max_rotate_angle > 0 || param_.max_shear_ratio > 0.0f
```

Comment:
this should be `else if`?

Reply:
It is possible that users want to add rotation on top of random resized crop.